qutip/stochastic.py

```diff
@@ -302,116 +548,123 @@ def ssesolve(H, psi0, times, sc_ops=[], e_ops=[], _safe_mode=True, **kwargs):
+
+    sso.LH = liouvillian(sso.H, c_ops=sso.sc_ops + sso.c_ops) * sso.dt
+    if sso.method == 'homodyne' or sso.method is None:
+        if sso.m_ops is None:
+            sso.m_ops = [op + op.dag() for op in sso.sc_ops]
+        sso.sops = [spre(op) + spost(op.dag()) for op in sso.sc_ops]
+        if not isinstance(sso.dW_factors, list):
+            sso.dW_factors = [1] * len(sso.m_ops)
+        elif len(sso.dW_factors) != len(sso.m_ops):
+            raise Exception("The len of dW_factors is not the same as m_ops")
+
+    elif sso.method == 'heterodyne':
+        if sso.m_ops is None:
+            m_ops = []
+        sso.sops = []
+        for c in sso.sc_ops:
+            if sso.m_ops is None:
+                m_ops += [c + c.dag(), -1j * c - c.dag()]
+            sso.sops += [(spre(c) + spost(c.dag())) / np.sqrt(2),
+                         (spre(c) - spost(c.dag())) * -1j / np.sqrt(2)]
+        sso.m_ops = m_ops
+        if not isinstance(sso.dW_factors, list):
+            sso.dW_factors = [np.sqrt(2)] * len(sso.sops)
+        elif len(sso.dW_factors) == len(sso.m_ops):
+            pass
+        elif len(sso.dW_factors) == len(sso.sc_ops):
+            dW_factors = []
+            for fact in sso.dW_factors:
+                dW_factors += [np.sqrt(2) * fact, np.sqrt(2) * fact]
+            sso.dW_factors = dW_factors
+        elif len(sso.dW_factors) != len(sso.m_ops):
+            raise Exception("The len of dW_factors is not the same as sc_ops")
+
+    elif sso.method == "photocurrent":
+        raise NotImplementedError("Moved to 'photocurrentmesolve'")
 
-        else:
-            raise Exception("Unrecognized method '%s'." % sso.method)
+    else:
+        raise Exception("The method must be one of None, homodyne, heterodyne")
 
-    if sso.distribution == 'poisson':
-        sso.homogeneous = False
+    sso.ce_ops = [QobjEvo(spre(op)) for op in sso.e_ops]
+    sso.cm_ops = [QobjEvo(spre(op)) for op in sso.m_ops]
 
-    if sso.solver == 'euler-maruyama' or sso.solver is None:
-        sso.rhs = _rhs_psi_euler_maruyama
+    sso.LH.compile()
+    [op.compile() for op in sso.sops]
+    [op.compile() for op in sso.cm_ops]
+    [op.compile() for op in sso.ce_ops]
 
-    elif sso.solver == 'platen':
-        sso.rhs = _rhs_psi_platen
+    if sso.solver_code in [103, 153]:
+        sso.imp = 1 - sso.LH * 0.5
+        sso.imp.compile()
 
-    else:
-        raise Exception("Unrecognized solver '%s'." % sso.solver)
+    sso.solver_obj = sme
+    sso.solver_name = "smesolve_" + sso.solver
 
-    res = _ssesolve_generic(sso, sso.options, sso.progress_bar)
+    res = _sesolve_generic(sso, sso.options, sso.progress_bar)
 
     if e_ops_dict:
         res.expect = {e: res.expect[n]
                       for n, e in enumerate(e_ops_dict.keys())}
-
     return res
 
 
-def smesolve(H, rho0, times, c_ops=[], sc_ops=[], e_ops=[],
-            _safe_mode=True ,**kwargs):
+def ssesolve(H, rho0, times, sc_ops=[], e_ops=[],
```

Comment:
```suggestion
def ssesolve(H, psi0, times, sc_ops=[], e_ops=[],
```
If this only accepts ket, then this more clear